lib/acts_as_paranoid/core.rb

```diff
@@ -112,10 +118,14 @@ def destroy
           end
         end
       else
-        destroy!
+        destroy_fully!
       end
     end
 
+    def destroy
+      destroy!
```

Comment:
@jodosha the reason is that ActiveRecord currently assumes that `#destroy` and `#destroy!` do basically the same thing, except the latter raises exceptions. Because of this, it thinks it's safe to call `#destroy!` on dependent objects when the user calls `#destroy` on the parent object. However, AAP uses the `!` to signify a hard delete. I resolved this conflict by implementing the `#destroy_fully` method. See also the commit comment here: 3c12279a093eeebe22596a9db92eacb19ca5e62d